WcaOnRails/app/models/user.rb

```diff
@@ -293,6 +285,44 @@ def after_confirmation
     end
   end
 
+  def software_team?
+    team_member?('software') != false
+  end
+
+  def results_team?
+    team_member?('results') != false
+  end
+
+  def wrc_team?
+    team_member?('wrc') != false
+  end
+
+  def wdc_team?
+    team_member?('wdc') != false
+  end
+
+  def team_member?(team)
+    member = self.team_members.find_by_team_id( self.teams.find_by_friendly_id(team) )
+    if member && (member.end_date == nil || member.end_date >= Date.today)
+      return member
+    else
+      return false
+    end
+  end
+
+  def was_team_member?(team)
```

Comment:
Same comments as above: adding a method to team_members seems like it would clean this up. It also looks like this method only looks at one of the team_members, rather than all of them.